### PR TITLE
Proposal for #505

### PR DESCRIPTION
**Pull request by thomaschiozzi-tndigit**
Proposal for enhancement described in #505
Added a new field to identity provider and attribute provider settings map in order to store internal notes. Updated console to reflect those changes.

### CODE REVIEW

File: src/main/java/it/smartcommunitylab/aac/attributes/provider/AttributeProviderSettingsMap.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public class AttributeProviderSettingsMap extends AbstractSettingsMap {
     private Set<String> attributeSets;
     private PersistenceMode persistence;
     private String events;
+    private String internalNotes;
```
Comment (matteo-s) [inline src/main/java/it/smartcommunitylab/aac/attributes/provider/AttributeProviderSettingsMap.java:44]: rename to `notes`

File: src/main/java/it/smartcommunitylab/aac/identity/provider/IdentityProviderSettingsMap.java [hunk 2]
```diff
@@ -48,6 +48,7 @@ public class IdentityProviderSettingsMap extends AbstractSettingsMap {
     private String events;
     private Integer position;
     private String template;
+    private String internalNotes;
```
Comment (matteo-s) [inline src/main/java/it/smartcommunitylab/aac/identity/provider/IdentityProviderSettingsMap.java:51]: rename to `notes`

**Review by matteo-s — COMMENTED** [review]
In order to add a user-submitted string to properties you need to properly process the value to avoid xss/script injection.
Edit the controllers and process with Jsoup safelist the content, if non-null.

Example:
```
@PostMapping("/idps/{realm}")
    @Operation(summary = "add a new identity provider to a given realm")
    public ConfigurableIdentityProvider addIdp(
        @PathVariable @Valid @NotNull @Pattern(regexp = SystemKeys.SLUG_PATTERN) String realm,
        @RequestBody @Valid @NotNull ConfigurableIdentityProvider reg
    )
        throws NoSuchRealmException, NoSuchProviderException, RegistrationException, SystemException, NoSuchAuthorityException {
        logger.debug("add idp to realm {}", StringUtils.trimAllWhitespace(realm));

        // enforce realm match
        reg.setRealm(realm);

        String notes = reg.getNotes();
        if (StringUtils.hasText(notes)) {
            notes = Jsoup.clean(notes, Safelist.none());
        }
        reg.setNotes(notes);

        if (logger.isTraceEnabled()) {
            logger.trace("idp bean: {}", StringUtils.trimAllWhitespace(reg.toString()));
        }

        return providerManager.addProvider(realm, reg);
    }

```

**Comment by thomaschiozzi-tndigit** [comment]
The sanitization task proposed in https://github.com/scc-digitalhub/AAC/pull/506#pullrequestreview-1821561426 has been broken down into 2 parts:
(1) the introduction of a new annotation for constrained string fields, where the constraint is satisfied if and only if given safety requirement are satisifed (currently mostly based on the Jsoup safelist, see [reference link](https://jsoup.org/apidocs/org/jsoup/safety/Safelist.html) - but it can possibly be expanded in the future if required)
(2) the actual input sanitization task, delegated to the corresponding provider service.

**Comment by thomaschiozzi-tndigit** [comment]
Closed after review. See #519 instead.

**Pull request closed without merging**